pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -209,27 +209,39 @@ private void applyQueryOptions(QueryContext queryContext) {
 
     // Set group-by query options
     if (QueryContextUtils.isAggregationQuery(queryContext) && queryContext.getGroupByExpressions() != null) {
-
       // Set maxInitialResultHolderCapacity
-      queryContext.setMaxInitialResultHolderCapacity(_maxInitialResultHolderCapacity);
-
+      Integer initResultCap = QueryOptionsUtils.getMaxInitResultCap(queryOptions);
+      if (initResultCap != null) {
+        queryContext.setMaxInitialResultHolderCapacity(initResultCap);
+      } else {
+        queryContext.setMaxInitialResultHolderCapacity(_maxInitialResultHolderCapacity);
+      }
```

Comment:
nit: unboxing can be coupled with default.

```suggestion
      Integer initResultCap = QueryOptionsUtils.getMaxInitResultCap(queryOptions);
      int maxInitialResultHolderCapacity = initResultCap == null ? _maxInitialResultHolderCapacity : initResultCap;
      queryContext.setMaxInitialResultHolderCapacity(maxInitialResultHolderCapacity);
```

Reply:
Done